apps/jobs/utils.py

```diff
@@ -222,3 +226,142 @@ def handle_submission_rerun(submission, updated_status):
                 "submitted_image_uri"
             ]
     return message
+
+
+def calculate_distinct_sorted_leaderboard_data(
```

Comment:
Can you please add numpy based docstring https://numpydoc.readthedocs.io/en/latest/format.html#id8?